hs_core/hydroshare/resource.py

```diff
@@ -876,6 +876,58 @@ def delete_resource(pk):
     return pk
 
 
+def get_resource_file_name(f):
+    '''
+    get the file name of a specific ResourceFile object f
+    Args:
+        f: the ResourceFile object to return name for
+    Returns:
+        the file name of the ResourceFile object f
+    '''
+    if f.resource_file:
+        # file was originally from local disk, and is currently stored in irods hydroshare zone
+        res_fname = f.resource_file.name
+    elif f.fed_resource_file:
+        # file was originally from local disk, but is currently stored in federated irods zone
+        res_fname = f.fed_resource_file.name
+    else:
+        # file was originally from federated irods zone, and is currently stored in the
+        # same federated irods zone
+        res_fname = f.fed_resource_file_name_or_path
+    return res_fname
+
+
+def delete_resource_file_only(resource, f):
```

Comment:
Though this function works for now, and I am OK with it, I am making a note here for possible further improvement at a later stage. I would preferably move this method as a classmethod of the ```ResourceFile``` class and rename the method as ```delete(f)```. Then this can be used as ```ResourceFile.delete(f)```.